### PR TITLE
fix(plugin-version): report the dependent workspace when unable to upgrade

### PR DESCRIPTION
**Pull request by Samantha-uk**
**What's the problem this PR addresses?**
When running yarn version <strategy>, it would incorrectly report the name of any dependent packages it was unable to auto-upgrade.  The package name displayed was the name of the workspace being bumped rather than the name of the dependent.

**How did you fix it?**
Following the expert guidance of @merceyz  I changed 
```        report.reportWarning(MessageName.UNNAMED, `Couldn't auto-upgrade range ${range} (in ${structUtils.prettyLocator(project.configuration, workspace.anchoredLocator)})`);```
to
```        report.reportWarning(MessageName.UNNAMED, `Couldn't auto-upgrade range ${range} (in ${structUtils.prettyLocator(project.configuration, dependent.anchoredLocator)})`);```

**Checklist**

- [x] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).

- [x] I have set the packages that need to be released for my changes to be effective.

- [x] I will check that all automated PR checks pass before the PR gets reviewed.

### CODE REVIEW

**Comment by merceyz** [comment]
Could you add a test to https://github.com/yarnpkg/berry/blob/5f0439a087b8463d97fb70c96dc6f983a91be512/packages/acceptance-tests/pkg-tests-specs/sources/commands/version.test.js so we prevent regressions?

**Comment by Samantha-uk** [comment]
> Could you add a test to https://github.com/yarnpkg/berry/blob/5f0439a087b8463d97fb70c96dc6f983a91be512/packages/acceptance-tests/pkg-tests-specs/sources/commands/version.test.js so we prevent regressions?

Test added :)

File: packages/acceptance-tests/pkg-tests-specs/sources/commands/version.test.js [hunk 1]
```diff
@@ -1,5 +1,7 @@
+import {xfs, ppath} from '@yarnpkg/fslib';
+
 const {
-  fs: {readJson},
+  fs: {readJson,writeFile},
```
Comment (merceyz) [inline packages/acceptance-tests/pkg-tests-specs/sources/commands/version.test.js:4]:
```suggestion
  fs: {readJson},
```

File: packages/acceptance-tests/pkg-tests-specs/sources/commands/version.test.js [hunk 2]
```diff
@@ -196,5 +198,71 @@ describe(`Commands`, () => {
...
+          await xfs.writeJsonPromise(ppath.join(pkgDependant, `package.json`), {
+            name: `pkg-dependant`,
+            version: `1.0.0`,
+            dependencies: {
+              [`pkg-primary`]: `workspace:*`,
+            },
+          });
+
+          // Ensure we have the appropiate plugin.
+          await writeFile(`${path}/.yarnrc.yml`, [
```
Comment (merceyz) [inline packages/acceptance-tests/pkg-tests-specs/sources/commands/version.test.js:233]:
```suggestion
          await xfs.writeFilePromise(`${path}/.yarnrc.yml`, [
```

File: packages/acceptance-tests/pkg-tests-specs/sources/commands/version.test.js [hunk 3]
```diff
@@ -196,5 +198,71 @@ describe(`Commands`, () => {
...
+          let stdout;
+          let stderr;
+          let code;
+
+          try {
+            // Ensure everything is in place.
+            await run(`install`);
+
+            // Execute a version patch.
+            ({code, stdout, stderr} = await run(`workspace`,`pkg-primary`,`version`, `patch`));
```
Comment (merceyz) [inline packages/acceptance-tests/pkg-tests-specs/sources/commands/version.test.js]: Think we need to update the eslint config a bit cc @paul-soporan

Comment (paul-soporan) [reply packages/acceptance-tests/pkg-tests-specs/sources/commands/version.test.js]: #2293

Comment (merceyz) [reply packages/acceptance-tests/pkg-tests-specs/sources/commands/version.test.js]: I enabled some more rules in https://github.com/yarnpkg/berry/pull/2293 as well

**Review by merceyz — APPROVED** [review]

**Comment by merceyz** [comment]
Thanks!

**Pull request merged**